### PR TITLE
[API] Console Application for Searching Wikipedia.

### PR DESCRIPTION
**Pull request by Najlaalrajhia98**

### CODE REVIEW

**Review by BaraahAlKalbani — APPROVED** [review]

**Review by BaraahAlKalbani — APPROVED** [review]

File: src/main/java/org/example/WikipediaSearch.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                    JsonArray searchResults = responseJson.getAsJsonObject("query").getAsJsonArray("search");
+
+                    /**
+                     * @for-each used to iterate over the search results, extracts the title and snippet for each search result, and prints them to the console
+                     * the search results are represented as array of elements of json objects so to extract properties that are inside the objects we need to first cast it to a json object
+                     */
+                    for (JsonElement result : searchResults) {
+                        JsonObject resultObject = result.getAsJsonObject();
+                        String title = resultObject.get("title").getAsString();
+                        String snippet = resultObject.get("snippet").getAsString().replaceAll("<.*?\\>","");
```
Comment (CodelineAtyab) [inline src/main/java/org/example/WikipediaSearch.java:78]: Please put a detailed comment that explains what this regular expression is doing. LIke what do we mean by <.*?\\>

File: src/main/java/org/example/WikipediaSearch.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                    .build();
+
+            try (Response response = client.newCall(request).execute()) {
+
+                if (response.isSuccessful()) {
+                    // Extract the response body as a String
+                    String responseBody = response.body().string();
+
+                    // Using a JSON parsing library allows the code to easily extract the relevant data from the API response and work with it in a structured way.
+                    JsonObject responseJson = JsonParser.parseString(responseBody).getAsJsonObject();
```
Comment (CodelineAtyab) [inline src/main/java/org/example/WikipediaSearch.java:61]: Please mention the comment explaining how this JsonObject JsonArray and JsonElement works together.

**Review by CodelineAtyab — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/WikipediaSearch.java (hunk 1 above)
Comment (Najlaalrajhia98) [reply src/main/java/org/example/WikipediaSearch.java:78]: done

File: src/main/java/org/example/WikipediaSearch.java (hunk 2 above)
Comment (Najlaalrajhia98) [reply src/main/java/org/example/WikipediaSearch.java:61]: done

**Review by FatemaAlbalushi — APPROVED** [review]
Nice work

**Pull request merged**